simphony/testing/abc_check_engine.py

```diff
@@ -35,46 +40,48 @@ def create_dataset_items(self):
         """
 
     @abc.abstractmethod
-    def check_emtpy_dataset(self, ds):
-        """ Check if a dataset is empty
-        """
-
-    @abc.abstractmethod
     def check_instance_of_dataset(self, ds):
         """ Check if a dataset is instance of a class
         """
 
-    @abc.abstractmethod
-    def check_is_usable_dataset(self, ds, items):
-        """ Check if a dataset is usable
-        """
+    operation_mapping = {
+        'compare datasets': 'none',
+        'add item': 'none'}
+
+    def compare_operation(self, *args, **kwrds):
+        method = getattr(
+            sys.modules[__name__],
+            self.operation_mapping['compare datasets'])
+        return method(*args, **kwrds)
+
+    def add_operation(self, container, *args, **kwrds):
+        method = getattr(
+            sys.modules[__name__],
+            self.operation_mapping['add item'])
+        return method(*args, **kwrds)
 
     def test_get_missing_dataset(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
+        with closing(self.engine_factory()) as handle:
             with self.assertRaises(ValueError):
                 handle.get_dataset('foo')
 
     def test_add_dataset_empty(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
-            handle.add_dataset(self.create_dataset(name='test'))
+        with closing(self.engine_factory()) as handle:
```

Comment:
> be correct? I supose that we have to provide this if we want to tests the FILE-IO wrappers. That's why i left the closing there in the first place. What do you think?

Any specific cleanup can just be handled in the `tearDown` method of the classes subclassing `CheckEngine`.  So an additonal function 'engine_close' isn't required.

All our wrappers (regardless of being FILE-IO/INTERNAL) implement the ABCModelingEngine interface and with this framework can then be tested in the same manner.   For example, in simphony-lammps-md, we have both a FILE-IO and INTERNAL wrapper and both are tested in exactly the same manner: https://github.com/simphony/simphony-lammps-md/blob/master/simlammps/tests/test_lammps_wrapper.py and https://github.com/simphony/simphony-lammps-md/blob/master/simlammps/testing/abc_lammps_md_engine_check.py.   Once 0.2.0 of simphony-common is released, those tests in simphony-lammps will be changed/extended to use the testing frame work you are developing (i.e. CheckEngine, etc)